### PR TITLE
Handle empty bytecode case in `get_code`

### PR DESCRIPTION
**Pull request by tcoratger**
Time spent on this PR: 1h

Resolves: #859 

# Pull Request type

Please check the type of change your PR introduces:

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?

- The case where the bytecode is null is handled in `get_code`. Indeed, when the bytecode is null (and after the other conditions are met), instead of outputting an error, we can simply call the function which gives us the length of the bytecode. If this is zero then we know that the bytecode is null and we can return the default value (as in the case where the contract does not exist).
- A unit test is used to test this functionality with a null bytecode contract deployed for this occasion.

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by greged93** [comment]
Ah sorry should have added this to the issue, this is a bug in cainome, raised an issue here, hopefully fixed soon (you can also PR on cainome if you want)

https://github.com/cartridge-gg/cainome/issues/24

File: tests/eth_provider.rs [hunk 1]
```diff
@@ -218,6 +218,25 @@ async fn test_get_code(#[future] counter: (Katana, KakarotEvmContract), _setup:
...
+    let eth_provider = katana.eth_provider();
+    let counter_address: Felt252Wrapper = counter.evm_address.into();
+    let counter_address = counter_address.try_into().expect("Failed to convert EVM address");
+
+    // When
+    let bytecode = eth_provider.get_code(counter_address, None).await.unwrap();
+
+    // Then
+    assert_eq!(bytecode, Bytes::default());
+}
```
Comment (greged93) [inline tests/eth_provider.rs:238]: nice!

File: src/eth_provider/provider.rs [hunk 2]
```diff
@@ -350,7 +350,11 @@ where
         let account_contract = AccountContractReader::new(address, &self.starknet_provider);
         let bytecode = account_contract.bytecode().block_id(starknet_block_id).call().await;
 
-        if contract_not_found(&bytecode) || entrypoint_not_found(&bytecode) {
+        if contract_not_found(&bytecode)
+            || entrypoint_not_found(&bytecode)
+            || account_contract.bytecode_len().block_id(starknet_block_id).call().await.map_err(KakarotError::from)?.len
+                == FieldElement::ZERO
+        {
```
Comment (greged93) [inline src/eth_provider/provider.rs:357]: as discussed let's just replace this with a call to `self.starknet_provider.call(...)`

Comment (tcoratger) [reply src/eth_provider/provider.rs]: Just fixed..

**Review by greged93 — APPROVED** [review]

**Pull request merged**